### PR TITLE
Fix tests failing on CUDA

### PR DESCRIPTION
**Pull request by neerajprad**
This fixes all tests failing on CUDA in preparation for a minor release. It also updates the torch requirement to `1.0.1` which seems to be needed by the `LKJCorrCholesky` distribution.

### CODE REVIEW

File: tests/infer/test_gradient.py [hunk 1]
```diff
@@ -70,10 +70,10 @@ def guide(subsample):
     inference = SVI(model, guide, optim, loss=elbo)
     with xfail_if_not_implemented():
         if subsample_size == 1:
-            inference.loss_and_grads(model, guide, subsample=torch.LongTensor([0]))
```
Comment (jpchen) [inline tests/infer/test_gradient.py]: is there a difference here? wont it default to the right device?

File: pyro/distributions/lkj.py [hunk 2]
```diff
@@ -202,9 +202,11 @@ def log_prob(self, x):
         Km1 = self._d - 1
 
         log_diagonals = x.diagonal(offset=0, dim1=-1, dim2=-2)[..., 1:].log()
+        # TODO: Figure out why the `device` kwarg to torch.linspace seems to not work in certain situations,
+        # and a seemingly redundant .to(x.device) is needed below.
```
Comment (jpchen) [inline pyro/distributions/lkj.py]: shouldnt we just remove the `device=x.device` then?

File: tests/infer/test_gradient.py (hunk 1 above)
Comment (neerajprad) [reply tests/infer/test_gradient.py]: Its really odd, but it doesn't:

```python
In [1]: import torch

In [2]: torch.set_default_tensor_type('torch.cuda.DoubleTensor')

In [3]: torch.LongTensor(1)
Out[3]: tensor([140263273888632], device='cpu')

```

File: pyro/distributions/lkj.py (hunk 2 above)
Comment (neerajprad) [reply pyro/distributions/lkj.py]: The device arg to `torch.linspace` should work fine, and I'm only facing this issue when I run `test_cuda.py`. Having the `.to` should be redundant in most cases, but removing the `device` arg will be 2-3 times slower.

**Comment by neerajprad** [comment]
Thanks for reviewing, @jpchen and @fritzo.

**Pull request merged**